tools/export_to_path/export_to_path.py

```diff
@@ -15,38 +14,47 @@
 parser = optparse.OptionParser()
 parser.add_option('-d', '--export_dir', help='Directory where to export the datasets')
 (options, args) = parser.parse_args()
-if not options.export_dir:
-    parser.error('Export directory cannot be empty')
-
-dir_prefix = "/mnt/sally"
-dir_suffix = 'mzML_profile'
-full_export_dir = os.path.join(dir_prefix, options.export_dir.lstrip(os.sep), dir_suffix)
-real_export_dir = os.path.realpath(full_export_dir)
-if not real_export_dir.startswith(dir_prefix):
-    raise Exception("'%s' must be a subdirectory of '%s'" % (real_export_dir, dir_prefix))
-if not os.path.exists(real_export_dir):
-    os.makedirs(real_export_dir)
+
+
+def get_path():
+    dir_prefix = '/mnt/sally'
+    dir_suffix = 'mzML_profile'
+    if not options.export_dir:
+        parser.error('Input cannot be empty')
+    elif not os.path.isdir(os.path.join(dir_prefix, options.export_dir.lstrip(os.sep))):
+        parser.error('Subdirectory must exist')
+    else:
+        return os.path.join(dir_prefix, options.export_dir.lstrip(os.sep), dir_suffix)
+
+
+export_path = get_path()
+
+
+if not os.path.exists(export_path):
+    os.makedirs(export_path)
 
 dataset_paths = args[::3]
 dataset_names = args[1::3]
-dataset_exts = args[2::3]
+dataset_extensions = args[2::3]
+
 exit_code = 0
-for dp, dn, de in zip(dataset_paths, dataset_names, dataset_exts):
-    """
-    Copied from get_valid_filename from django
-    https://github.com/django/django/blob/master/django/utils/text.py
-    """
+
+for dp, dn, de in zip(dataset_paths, dataset_names, dataset_extensions):
+    '''
+    Copied from django https://github.com/django/django/blob/master/django/utils/text.py
+    '''
     dn_de = "%s.%s" % (dn, de)
     dn_de_safe = re.sub(r'(?u)[^-\w.]', '', dn_de.strip().replace(' ', '_'))
-    dest = os.path.join(real_export_dir, dn_de_safe)
-    if os.path.isfile(dest):
-        raise Exception("Error copying dataset '%s' to '%s'. Cannot overwrite existing dataset" % (dn, dest))
+
+    if os.path.isfile(os.path.join(export_path, dn_de_safe)):
```

Comment:
I'd argue that collapsing this into a single line is harder to read than before.